### PR TITLE
 Suppress unnecessary warnings about ANSICON in integrated terminal of vscode

### PR DESCRIPTION
**Pull request by yosukei3108**
# Description
On the integrated terminal of vscode, Cucumber always warns about ANSICON as below.

```bash
*** WARNING: You must use ANSICON 1.31 or higher (https://github.com/adoxa/ansicon/) to get coloured output on Windows
```

This behavior was discussed in https://github.com/cucumber/cucumber-ruby/issues/1163 or https://github.com/cucumber/cucumber-ruby/pull/1119, but unfortunately has not resolved yet.

Because the environment variable `VSCODE_PID` mentioned in #1163 is already  obsolete, it seems to be good to check the value of `TERM_PROGRAM` instead of it.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Update required of cucumber.io/docs

# Checklist:

Your PR is ready for review once the following checklist is
complete. You can also add some checks if you want to.

- [x] Tests have been added for any changes to behaviour of the code
  - I have tested the above change with my local copy of the gem, and I am able to see the correct color output from cucumber in the vscode integrated terminal.
- [x] New and existing tests are passing locally and on CI
- [x] `bundle exec rubocop` reports no offenses
- [x] (N/A) RDoc comments have been updated
- [x] (N/A) CHANGELOG.md has been updated

### CODE REVIEW

**Comment by aurelien-reeves** [comment]
Thanks for your PR!

Myself - as Windows 10 now supports AINSI colors for the terminal - I would be in favor of totally remove that auto-discovery attempt of coloring support for Windows and to keep colors enabled per default.

For older version of Windows, I would instead document the fact that colors may be unsupported and that user should turn them off themselves using the `--no-color` flag.

@cucumber/cucumber-ruby  what do you think?

**Comment by aurelien-reeves** [comment]
@yosukei3108 would you feel confident in:
- removing the `if windows` block in `ainsicolor.rb`
- updating the changelog explaining that support for automatic detection of ansicon on Windows has been removed
- creating a new document in `docs` explaining to legacy windows users that coloring may be broken, that they may need to use `--no-color` flag, or to install ANSICON (with a link to the ansicon repo)?

**Comment by yosukei3108** [comment]
@aurelien-reeves thanks for your comments!

> I would be in favor of totally remove that auto-discovery attempt of coloring support for Windows and to keep colors enabled per default.

I also think that is much better.

> would you feel confident in:

Your kind hints would help me. I'll do my best!

**Comment by aurelien-reeves** [comment]
> Your kind hints would help me. I'll do my best!

:)

So to begin with, I guess we could simply remove the following block of code from `ansicolor.rb`

```ruby
if Cucumber::WINDOWS_MRI
  unless ENV['ANSICON'] || ENV['TERM_PROGRAM'] == 'vscode'
    STDERR.puts %{*** WARNING: You must use ANSICON 1.31 or higher (https://github.com/adoxa/ansicon/) to get coloured output on Windows}
    Cucumber::Term::ANSIColor.coloring = false
  end
end
```

Make sure tu execute `bundle exec rake` to execute all the tests, but I think it should be alright.

Then in the CHANGELOG.md, we should add an entry in the `In Git > removed` paragraph to tell that on Windows, support for ANSICON has been removed due to recent version of Windows properly supporting ansi colors in the terminal, with a note inviting users to fallback on using `--no-color` or installing ANSICON globally if needed (with a link to https://github.com/adoxa/ansicon/).

Actually I guess that this will be enough for now as the info will be at the top of the changelog.
If we notice that some user are asking about that, we may have a dedicated doc, but for now it won't be necessary.

@yosukei3108 does that sounds good?

File: CHANGELOG.md
Comment (aurelien-reeves) [inline CHANGELOG.md:33]:
```suggestion
- On Windows, auto-detection of ANSICON has been removed - Windows now properly supports ANSI colors in the terminal. In case of issues on older version of Windows, execute cucumber with `--no-color`, or install [ANSICON](https://github.com/adoxa/ansicon/) globally.
```

**Review by aurelien-reeves — COMMENTED** [review]
I suggest some changes in the changelog. But if a native English speaker could make sure it makes sense, that could be great 😅

File: CHANGELOG.md
Comment (yosukei3108) [reply CHANGELOG.md]: Thank you for your suggestion and sorry my poor English...
And I know `Commit suggestion` in Github just now, after I committed the correction manually 😭

**Comment by aurelien-reeves** [comment]
Great @yosukei3108, thank you!
Just wait for another review, but it looks good to me 👍

File: CHANGELOG.md
Comment (aurelien-reeves) [reply CHANGELOG.md]: Don't apologize! My English is not good either

No worries for the suggestion commit 😁

**Review by mattwynne — APPROVED** [review]

**Comment by aslakhellesoy** [comment]
Hi @yosukei3108,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**
